include/dirent.h

```diff
@@ -35,7 +44,47 @@ extern DIR *fdopendir(int fd) __DARWIN_ALIAS(fdopendir);
 extern DIR *fdopendir(int fd) __DARWIN_ALIAS_I(fdopendir);
 #endif
 
-__MP__END_DECLS
+/* Wrapper functions/macros to support fdopendir */
+extern DIR *__mpls_opendir(const char *name);
+extern int __mpls_closedir(DIR *dirp);
+extern int __mpls_dirfd(DIR *dirp);
+
+#define opendir(name)      __mpls_opendir(name)
+#define closedir(dirp)     __mpls_closedir(dirp)
+
+#ifndef __MP_LEGACY_SUPPORT_NO_DIRFD_MACRO
+#undef dirfd(dirp)
```

Comment:
This should be just
```
#undef dirfd
```
otherwise warnings

Reply:
Thanks. It's done.